### DISCUSSION TITLE
Обработчик ошибок

### DISCUSSION THREAD

**Discussion by forwork10** [Sokol-Games-GmbH/PixelCombats.PublicApi#112] [Ideas]
добавить класс для обработки ошибок во время выполнения кода, например:

```javascript
RuntimeErrorHandler.Add(function(e){
msg.ShowError();
});
```
сейчас при любой ошибке внутри функции она просто завершает выполнение, а прописывать везде конструкции try - catch лень

**Comment by kkohno** [comment]
Не совсем корректная задача. 
Такого не видел ни разу и не возникало проблем с этим.
Однако, если уж лень то cделайте свою обработку средствами JavaScript

всегда можно написать так:
```javascript
function my_function(){
    try{
        // тело функции
    }
    catch(){
        // обработка ошибки
    }
}
```
или написать что-то подобное:
```javascript
// то, что вызывать будем
function my_function(){
    // тело функции
}
// свой обработчик
function executor(func){
    try{
        // тут вызвать функцию func
       func();
    }
    catch(){
        // обработка ошибки
    }
}

// вызов функции
executor(my_function);
```

**Comment by forwork10** [comment]
как тогда по вашему различные языки программирования выводят ошибки во время выполнения кода?
